spec/components/single_page_notification_button_spec.rb

```diff
@@ -40,4 +40,53 @@ def component_name
     render_component({ base_path: "/the-current-page", margin_bottom: 9 })
     assert_select '.gem-c-single-page-notification-button.govuk-\!-margin-bottom-9'
   end
+
+  it "has a data-module attribute for JavaScript, if the js-enhancement flag is present" do
+    render_component({ base_path: "/the-current-page", js_enhancement: true })
+    dom = Nokogiri::HTML(rendered)
+    form_data_module = dom.xpath("//form")[0].attr("data-module")
```

Comment:
Curious why Nokogiri is needed here - is it not possible to do something like `assert_select ".gem-c-single-page-notification-button[data-module="single-page-notification-button gem-track-click"]`?

Reply:
Yep it is possible to have `assert_select ".gem-c-single-page-notification-button[data-module="gem-track-click single-page-notification-button"]` and the test will pass. But just because something is possible, doesn't mean it's the best/most accurate way.

Change the order of the `data-module`s, or add another `data-module` to the component and this test will fail. IMO the test should be strictly scoped to the relevant thing (which in this case is `data-module` value of `single-page-notification-button`). But I can be convinced to change it if there's some major disadvantage to using the Nokogiri approach? 🤷🏻‍♀️ 
Admittedly I am no expert in these things.

Reply:
No, that's a good answer, thanks.